arrayfire/array.py

```diff
@@ -1185,20 +1185,35 @@ def to_list(self, row_major=False):
         ct_array, shape = self.to_ctype(row_major, True)
         return _ctype_to_lists(ct_array, len(shape) - 1, shape)
 
-    def __repr__(self):
+    def to_string(self):
```

Comment:
@unbornchikken How about implementing the `display_dims_limit` functionality you mentioned, but it is disabled by default. i.e. the `print(arr)` works as long as we detect PyCharm or user manually disables it by setting the limit.

Disabling for Other IDEs can be enabled as requested.